app/controllers/api/users.php

```diff
@@ -1091,6 +1091,16 @@ function createUser(string $hash, mixed $hashOptions, string $userId, ?string $e
             }
         }
 
+        if (\strlen($password) === 0) {
+            $user
+                ->setAttribute('password', '')
+                ->setAttribute('passwordUpdate', DateTime::now());
+
+            $user = $dbForProject->updateDocument('users', $user->getId(), $user);
+            $queueForEvents->setParam('userId', $user->getId());
+            $response->dynamic($user, Response::MODEL_USER);
+        }
+
```

Comment:
setting the password to empty should not be allowed.

1. This is not how we should perform a password reset. We already have a procedure for password reset and its using tokens which is more secure and the right way.
2. We have another endpoint in the accounts service to reset password
3. This method is insecure and a security footgun. So its better we dont allow it in the first place

Reply:
@christyjacob4 as Matej mentioned, this is not a password reset feature, but an ability to have the flexibility to convert the account to whatever the dev wants it to be. Its performed only from the server api. 

No reason why we should allow a user to be without a name, email or password but won't allow it to be edited to the same state.